### PR TITLE
Auth issues

### LINKED ISSUE #876: Users Sign In As Apollos-Server
### Details

We are seeing some users sign in as the `apollos-server` user. Here's the whole story:

One of our staff members came to us and let us know that she was receiving notifications that she was being prayed for even though she had never entered any prayers. 

After some research, it was determined that this was happening when someone prayed for one particular prayer we had that was somehow tied to the `apollos-server` user. 

Looking in Rock, it appears that there are multiple devices now associated with the `apollos-server` user in the PersonalDevice table. These started appearing in our production Rock instance on 06.30 and have continued to be added. 

This tells me that we have people who are someone successfully signing in to the app as the `apollos-server` user instead of their own user account. Or that someone it is signing in to both their accounts and the server account. 

#### For Consideration
Per Conrad: "As part of this issue we'll want to take a deep dive into all non-`get` requests we make to the API and make sure we're passing correct user credentials and failing if there are none."

### Expected Behavior

The user should be signed in to their specific account only.

### Actual Behavior

Some people are being signed in as the `apollos-server` user.

### Possible Fix

No idea.

<details><summary>Additional Info</summary>

### Your Environment

- Environment name and version (e.g. Chrome 39, node.js 5.4):
- Operating System and version (desktop or mobile):
- Link to your project:

### Steps to Reproduce

I cannot personally reproduce, but we have at least one person who is receiving notifications for prayers they haven't put in. I can provide any Rock data or information necessary to help in figuring this out if needed.

### Stack Trace

No stack trace.

</details>

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

User's were occasionally getting logged in as the Apollos user. The issue was that if a user's cookie was bad (potentially expired by rock?) the auth token was still being sent to Rock. The auth token belongs to the apollos user, so the apollos user was being sent as the current user. 

### What design trade-offs/decisions were made?

n/a

We still need a PR that logs users out if they receive a AuthenticationError on the front end. 

### How do I test this PR?

@conrad-vanl tested via deploying this to Heroku

### What automated tests did you write?

Updated existing snapshots. 

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes #876 
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-data-connector-rock/src/auth/__tests__/__snapshots__/index.tests.js.snap [hunk 1]
```diff
@@ -114,7 +114,7 @@ exports[`Auth registers an auth token and passes the cookie on requests to rock
 Headers {
   Symbol(map): Object {
     "Authorization-Token": Array [
-      "some-rock-token",
+      "null",
```
Comment (conrad-vanl) [inline packages/apollos-data-connector-rock/src/auth/__tests__/__snapshots__/index.tests.js.snap]: this looks like it's sending a string `"null"`. Shouldn't the header just be removed altogether somehow? Did you test making a request to the Rock API with the `Authorization-Token` being set to `"null"`?

Comment (vinnyjth) [reply packages/apollos-data-connector-rock/src/auth/__tests__/__snapshots__/index.tests.js.snap]: I did, and it worked fine, but I had a valid user cookie. I haven't tested the "invalid user cookie with `"null"` token"

**Comment by vinnyjth** [comment]
Logging in and out of the app, and user specific features seem to work as before. I'm not certain this fixes #876 though, but I currently don't have a good way to test.

**Pull request merged**